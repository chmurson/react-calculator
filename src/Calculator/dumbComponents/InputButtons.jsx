import React from 'react'
import PropTypes from 'prop-types'
import { GridWithKeyMapButton, ButtonWithKeyMap } from './ButtonsWithKeyMap'

export function InputButton({ input, onClick, ...restOfProps }) {
  return (
    <ButtonWithKeyMap keyToMap={input} onClick={() => onClick(input)} {...restOfProps}>
      {input}
    </ButtonWithKeyMap>
  )
}

InputButton.propTypes = {
  input: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export function InputGridButton({ input, onClick, ...restOfProps }) {
  return (
    <GridWithKeyMapButton keyToMap={input} onClick={() => onClick(input)} {...restOfProps}>
      {input}
    </GridWithKeyMapButton>
  )
}

InputGridButton.propTypes = {
  input: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}
