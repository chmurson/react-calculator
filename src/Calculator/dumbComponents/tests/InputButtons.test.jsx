import React from 'react'
import { mount } from 'enzyme'
import { InputGridButton, InputButton } from '../InputButtons'

describe('InputGridButton', () => {
  let component
  beforeAll(() => {
    component = mount(
      <InputGridButton input="3" onClick={() => null} row={1} col={1}>
        text
      </InputGridButton>
    )
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})

describe('InputButton', () => {
  let component
  beforeAll(() => {
    component = mount(
      <InputButton onClick={() => null} input="4">
        text
      </InputButton>
    )
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
