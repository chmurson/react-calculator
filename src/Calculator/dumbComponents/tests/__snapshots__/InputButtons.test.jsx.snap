// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton matches snapshot 1`] = `
<InputButton
  input="4"
  onClick={[Function]}
>
  <withKeyMap(Button)}
    innerRef={[Function]}
    keyToMap="4"
    onClick={[Function]}
  >
    <Button
      buttonRef={[Function]}
      className=""
      onClick={[Function]}
      title="Shortcut: 4"
    >
      <button
        className="custom-button custom-button--regular"
        onClick={[Function]}
        tabIndex={-1}
        title="Shortcut: 4"
      >
        4
      </button>
    </Button>
  </withKeyMap(Button)}>
</InputButton>
`;

exports[`InputGridButton matches snapshot 1`] = `
<InputGridButton
  col={1}
  input="3"
  onClick={[Function]}
  row={1}
>
  <withGridCell(withKeyMap(Button)})
    col={1}
    keyToMap="3"
    onClick={[Function]}
    row={1}
    style={Object {}}
  >
    <withKeyMap(Button)}
      innerRef={[Function]}
      keyToMap="3"
      onClick={[Function]}
      style={
        Object {
          "gridColumn": 1,
          "gridRow": 1,
        }
      }
    >
      <Button
        buttonRef={[Function]}
        className=""
        onClick={[Function]}
        style={
          Object {
            "gridColumn": 1,
            "gridRow": 1,
          }
        }
        title="Shortcut: 3"
      >
        <button
          className="custom-button custom-button--regular"
          onClick={[Function]}
          style={
            Object {
              "gridColumn": 1,
              "gridRow": 1,
            }
          }
          tabIndex={-1}
          title="Shortcut: 3"
        >
          3
        </button>
      </Button>
    </withKeyMap(Button)}>
  </withGridCell(withKeyMap(Button)})>
</InputGridButton>
`;
